src/snowflake/cli/plugins/nativeapp/codegen/artifact_processor.py

```diff
@@ -74,9 +73,17 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 class ArtifactProcessor(ABC):
     def __init__(
         self,
-        na_project: NativeAppProjectModel,
+        package_name: str,
+        project_root: Path,
+        deploy_root: Path,
+        bundle_root: Path,
+        generated_root: Path,
```

Comment:
as discussed offline earlier, please don't go back to individual args. I gave a lot of context around the change in https://github.com/snowflakedb/snowflake-cli/pull/1262, feel free to take a look. I'd be OK with passing an object that encapsulates the source of these individual fields, but this is a step backwards that is likely to cause issues soon. In particular, I'm thinking that need a reliable way to find the manifest file, and this will differ in v1 vs v2.

Reply:
Also, NativeAppProjectModel is already an abstraction that encapsulates the project definition model, so I'm doubly confused about why we need this change in the first place.

Reply:
`NativeAppProjectModel` is not the right abstraction to support project definition v2: there might be multiple application packages, multiple bundles/stages, etc. That's the main motivation for this change.
What `NativeAppCompiler` and `ArtifactProcessor` really need is a subset of `NativeAppProjectModel` in the v1 flow. For v2 it's slightly different since we need to know which entity is being bundled, and possibly be a little smart about its dependencies.

Grouped the required values as a `BundleContext` data class. We might need to refactor further once we add definition v2 logic, but it should be enough as a first step.

Reply:
Sure, I understand that, nothing prevents you from reworking the abstraction. You can modify/rename `NativeAppProjectModel ` if you want. You can also create some kind of context object as I suggested in scrum yesterday. That new abstraction would expose the paths that are used now, the artifact rules, etc. In v2 it's going to be tied to a specific entity, so you can just go for that upfront and make it some kind entity-specific context. When v1 is used, then the app project is the entity. Same abstraction, just a special case of it. It would work nicely and be extensible in the future.